src/vs/vscode.proposed.d.ts

```diff
@@ -168,14 +168,19 @@ declare module 'vscode' {
 		availableFormats: ColorFormat[];
 
 		/**
+		 * Controls whether the color decorator is rendered.
+		 */
+		renderDecorator: boolean;
```

Comment:
Users need to have settings to control the visibility of color decorator, but I'd like to give the decision to extensions.

Right now, our internal CSS and JSON extensions have four settings for decorators rendering: 

![image](https://user-images.githubusercontent.com/876920/29533465-e2fbf9d8-8666-11e7-9fc8-ff40d639d69e.png)

For pure CSS/JSON, the setting can be a single boolean. In this case, having a setting in the core which can be language-wise is good enough.

However, if the document has embedded languages, like CSS+JS in HTML, turning on Color Decorators only for CSS sounds fair and to make that happen in the core, we need either scoping, or asking the extension in which range should we render the decorators. Considering we already ask extensions for color range, we can ask them whether to render the inline color decorator for a specific range or not at the same time.

If we decide to make ColorPicker and Color Decorators two standalone features, then I prefer to have a user setting to control Color Decorators, and do following
1. Configuration migration, from `css.colorDecorators.enable` to `"[css]": { "editor.colorDecorators": true }`.
2. Maybe need a new API to ask extensions for which range of document are allowed to show inline decorators and what's the proper value they stand for.

#2 will make it pretty similar to what we already have. So my take here is

* `DocumentColorProvider` is for Color related features, including Color Picker, inline Color Decorator and even more in the future.
* Give the decision to extensions: show inline Color Decorator or not, show Color Picker as Hoever or Peek, etc. Extensions decide what kind of settings they want to expose to users.

Look forward to your comments.